packages/flutter/lib/src/material/segmented_button.dart

```diff
@@ -0,0 +1,770 @@
+/// options.
+///
+/// The options are represented by segments described with [ButtonSegment]
+/// entries in the [segments] field. Each segment has a [ButtonSegment.value]
+/// that is used to indicated which segment's are selected.
+///
+/// The [selected] field is a set of the selected segments' values. This should
+/// be updated by the app in response to [onSelectionChanged] updates.
+///
+/// By default only a single segment can be selected (for mutually exclusive
+/// choices). This can be relaxed with the [multiSelectionEnabled] field.
+///
+/// Like [ButtonStyleButton]s the [SegmentedButton]'s visuals can be
+/// configured with a [ButtonStyle] [style] field. However, unlike other
+/// buttons, some of the style parameters are applied to the entire segmented
+/// button, and others are used for each of the segments.
+///
+/// By default a checkmark icon is used to show selected items. To configure
+/// this behavior, you can use the [showSelectedIcon] and [selectedIcon] fields.
+///
+/// Individual segments can be enabled or disabled with their
+/// [ButtonSegment.enabled] flag, but if [onSelectionChanged] field is null
+/// then the entire segmented button will be disabled, regardless of the
+/// individual segment settings.
+///
+/// See also:
+///
+///   * Material Design spec: <https://m3.material.io/components/segmented-buttons/overview>
+///   * [ButtonStyle], which can be used to configure the appearance of the button.
+///   * [Radio], an alternative way to present the user with a mutually exclusive set of options.
+///   * [FilterChip], [ChoiceChip], which can be used when you need to show more then five options.
+class SegmentedButton<T> extends StatelessWidget {
+  /// Creates a segmented button.
+  ///
+  /// [segments] must contain at least one segment, but it is recommended
+  /// to have two to five segments.
+  ///
+  /// If [onSelectionChanged] is null then the entire segemented button will
+  /// be disabled.
+  ///
+  /// By default [selected] must only contain one entry. However, if
+  /// [multiSelectionEnabled] is true then [selected] can contain multiple entries.
+  /// If [emptySelectionAllowed] is true, then [selected] can be empty.
+  const SegmentedButton({
+    super.key,
+    required this.segments,
+    required this.selected,
+    this.onSelectionChanged,
+    this.multiSelectionEnabled = false,
+    this.emptySelectionAllowed = false,
+    this.style,
+    this.showSelectedIcon = true,
+    this.selectedIcon,
+  })  : assert(segments != null),
+        assert(segments.length > 0),
+        assert(selected != null),
+        assert(selected.length > 0 || emptySelectionAllowed),
+        assert(selected.length < 2 || multiSelectionEnabled);
+
+  /// Descriptions of the segments in the button.
+  final List<ButtonSegment<T>> segments;
+
+  /// Set of [ButtonSegment.value]s that indicate which [segments] are selected.
+  final Set<T> selected;
```

Comment:
Mention that this parameter is required.

Reply:
I would mention that it is "required" here because you might not be looking at the constructor signature when you're looking at the API doc page for this parameter.  It's in the same vein as mentioning what the default value is.